docs(recommendations): update documentation for recommendations (#18)
algolia/react-recommendations: packages/react-recommendations/README.md

> > `Translations`

Reviewer: ```suggestion
> `RecommendationTranslations`
```

Author: Done in https://github.com/algolia/ui-components/pull/18/commits/a1991cfb5cb8bc7bd4502425134439493283c578

> > `Translations`

Reviewer: ```suggestion
> `RecommendationTranslations`
```

Author: Done in https://github.com/algolia/ui-components/pull/18/commits/a1991cfb5cb8bc7bd4502425134439493283c578

```ts
type RecommendationTranslations = Partial<{
  title: string;
  showMore: string;
}>;
```

##### `children`

> `(props: ChildrenProps) => JSX.Element`

<details>

<summary><code>ChildrenProps</code></summary>

```ts
type ChildrenProps<TObject> = {
  recommendations: TObject[];
  View(props: unknown): JSX.Element;
  translations: RecommendationTranslations;
};
```

</details>

Render function to modify the default rendering.

The default implementation is:

```js
function defaultRender(props) {
  if (props.recommendations.length === 0) {
    return null;
  }

  return (
    <section className="auc-Recommendations">
      {props.translations.title && <h3>{props.translations.title}</h3>}

      <props.View />
    </section>
  );
}
```

##### `view`

> `(props: ViewProps) => JSX.Element`

<details>

<summary><code>ViewProps</code></summary>

```ts
type ViewProps<
  TItem extends RecordWithObjectID,
  TTranslations extends Record<string, string>,
  TClassNames extends Record<string, string>
> = {
  items: TItem[];
  itemComponent({ item: TItem }): JSX.Element;

>   classNames?: Partial<TClassNames>;

Reviewer: We don't pass `classNames`.

Author: Make sense

Author: Done in https://github.com/algolia/ui-components/pull/18/commits/25fa8410a96e0521a21ab5738b56d938434a2622

>   classNames?: Partial<TClassNames>;

Reviewer: We can remove the `Partial`s here.

Author: Done in https://github.com/algolia/ui-components/pull/18/commits/0e8e81363ba4baa6c642e47992f3f1ce29391dfb

> [`SearchParameters`](https://www.algolia.com/doc/api-reference/search-api-parameters/) | defaults to `{ analytics: false, enableABTest: false }`

### `transformItems`

> `(Array<RecordWithObjectID<TItem>>) => Array<RecordWithObjectID<TItem>>`

<details>

<summary><code>RecordWithObjectID</code></summary>

```ts
type RecordWithObjectID<TItem> = TItem & {
  objectID: string;
};
```

</details>

> Receives the items, and is called before displaying them. Should return a new array with the same shape as the original array. Useful for mapping over the items to transform, and remove or reorder them.

Reviewer: We can move this function above `details`.

```suggestion
Function to transform the items retrieved by Algolia. It's useful to edit, add, remove or reorder them.
```

Author: Done in https://github.com/algolia/ui-components/pull/18/commits/3335abd6b66e9621bb5985eb477a8cf869c2d82b
